adr/ADR-32.md

```diff
@@ -33,8 +34,8 @@ Service configuration relies on the following:
 - `statsHandler` - an optional function that returns unknown data that can be
   serialized as JSON. The handler will be provided the endpoint for which it is
   building a `EndpointStats`
-- `endpoint` - a subject and a handler (effectively equivalent to a NATS
-  subscription)
+- `endpoints` - a collection of endpoints (name + handler) to be associated with the
+  service. Those can be either passed in configuration, or addded via a method on service.
```

Comment:
Can we just simplify this and only have it as part of the configuration of the service and not as something later? Are their use cases where they will make a root and then sometimes later change to endpoint? Or If they make endpoints, will they add another later?
What about instead of the ability to add and endpoint later (like when the service is already running), we just have the ability to turn on and off an already configured/running endpoint?

Reply:
From what I know, some languages which do not use callbacks for services, but rather iterators (e.g. Rust) do not have the option to easily store endpoints in config, so there it is easier to have it as a method on service interface. It is language specific I guess, for Go I'm leaving endpoint setup on `Config`.